providers/yaml/src/main/java/org/jboss/resteasy/plugins/providers/YamlProvider.java

```diff
@@ -52,9 +73,19 @@ public Object readFrom(Class<Object> type, Type genericType, Annotation[] annota
       try {
          LogMessages.LOGGER.debugf("Provider : %s,  Method : readFrom", getClass().getName());
          if (isValidInternalType(type)) {
-            return new Yaml().load(entityStream);
+            // Use the old behavior of trusting everything
+            if (isTypeCheckDisabled()) {
+               return new Yaml().loadAs(entityStream, type);
+            }
+            final BaseConstructor constructor;
+            if (genericType instanceof ParameterizedType) {
+               constructor = new TypeSafeConstructor((ParameterizedType) genericType, getClassLoader(type), allowedList);
+            } else {
+               constructor = new TypeSafeConstructor(type, getClassLoader(type), allowedList);
+            }
+            return new Yaml(constructor).loadAs(entityStream, type);
          } else {

```

Comment:
If we get here, we're bypassing the use of TypeSafeConstructor.  Why is that?

Reply:
Just because that is how it was and I didn't want to change the behavior too much. It also seems like the type itself is not constructed as it uses the actual return type or parameter.